# Data Manipulation Starter Notebook

For this core assignment , we will use insurance dataset.

Student Name:  

## Table of Contents


This notebook will be used for 3 Core assignments:
- [📝 Loading Data (Practice)](#loading)
- [📝 Slicing (Practice)](#slicing)
- [📝 Filtering (Practice)](#filtering)


Click on the assignment names above to jump to the corresponding headers on Google Colab.  



___


<a name="loading"> </a>
# 📝 Loading Data (Practice)

1. Find and copy the file path of the insurance_mod.csv file

In [13]:
# Surround the path with quotes
fpath="C://Users/USER/Downloads/insurance_mod.csv"


In [14]:
import numpy as np
import pandas as pd

4) Load in the file as a Pandas dataframe named df_insurance


In [16]:
# Include needed import(s)
df_insurance=pd.read_csv('insurance_mod.csv')

# Load in the file as a Pandas dataframe named df_insurance


5) Preview the first 5 rows with .head()



In [18]:
# Preview the first 5 rows with .head()
df_insurance.head()

age     sex     bmi  children  smoker     region  charges
0   19  female  27.900         0       1  southwest  16885.0
1   18    male  33.770         1       0  southeast   1726.0
2   28    male  33.000         3       0  southeast   4449.0
3   33    male  22.705         0       0  northwest  21984.0
4   32    male  28.880         0       0  northwest   3867.0

6) Preview the last 5 rows with .tail()

In [20]:
# Preview the last 5 rows with .tail()
df_insurance.tail()

age     sex    bmi  children  smoker     region  charges
1333   50    male  30.97         3       0  northwest  10601.0
1334   18  female  31.92         0       0  northeast   2206.0
1335   18  female  36.85         0       0  southeast   1630.0
1336   21  female  25.80         0       0  southwest   2008.0
1337   61  female  29.07         0       1  northwest  29141.0

7) Obtain summary information with .info()

In [22]:
# Obtain summary information with .info()
df_insurance.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   int64  
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(3), object(2)
memory usage: 73.3+ KB


8) Use the output of .info() to answer the following in a text cell below:




a) How many rows are there?

- Answer: 1338

b) How many columns are there?

- Answer: 7

## You have completed the "Loading Data" Practice Assignment.


<a name="slicing"> </a>
# 📝 Slicing (Practice)


1) Slice the row for the customer at index 50 as a Series

In [28]:
# Use single brackets so the output is a Pandas Series
customer_row=df_insurance.iloc[50]
customer_row

age                18
sex            female
bmi            35.625
children            0
smoker              0
region      northeast
charges        2211.0
Name: 50, dtype: object

2) Slice the row for the customer at index 50 as a DataFrame

In [30]:
# Use double brackets so the output is a Pandas Dataframe
customer_row_df=df_insurance.iloc[[50]]
customer_row_df

age     sex     bmi  children  smoker     region  charges
50   18  female  35.625         0       0  northeast   2211.0

3) Find out if the customer at index 20 was a male or female (by slicing index 20 and column "Sex")

In [32]:
# find out if the customer at index 20 was a male or female by slicing index 20 and column "Sex"
customer_20=df_insurance.iloc[[20]][['sex']]
customer_20

sex
20  female

4) Slice the entire smoker column using square brackets


In [34]:
# Slice the entire smoker column using square brackets
df_insurance[['smoker']]

smoker
0          1
1          0
2          0
3          0
4          0
...      ...
1333       0
1334       0
1335       0
1336       0
1337       1

[1338 rows x 1 columns]

5) Slice the entire smoker column using .loc


In [36]:
# Slice the entire smoker column using .loc
df_insurance.loc[:,'smoker']


0       1
1       0
2       0
3       0
4       0
       ..
1333    0
1334    0
1335    0
1336    0
1337    1
Name: smoker, Length: 1338, dtype: int64

6) Slice the entire smoker column using either method, but this time save it as a variable called "smoker"


In [38]:
# Slice the entire smoker column using either method, but this time save it as a variable called "smoker"
smoker=df_insurance['smoker']
smoker

0       1
1       0
2       0
3       0
4       0
       ..
1333    0
1334    0
1335    0
1336    0
1337    1
Name: smoker, Length: 1338, dtype: int64


- Calculate the sum of the smoker variable to see how many of the customers smoke


In [40]:
# To see how many of the customers smoke, get the sum
smoker.sum()

274


- Calculate the mean of the smoker variable using .mean() to see the ratio of smokers in the dataset


In [42]:
# To get the ratio use .mean()
smoker.mean()

0.20478325859491778

<a name="filtering"> </a>
# 📝 Filtering (Practice)

1) Create a filter variable called under_30_filter for rows where the age is less than 30



In [119]:
# Create a filter called under_30_filter for the rows where the age is less than 30
#under_30_filter=df_insurance[['age']]<30
#type(under_30_filter)

In [121]:
under_30_filter=df_insurance['age']<30
type(under_30_filter)

pandas.core.series.Series

In [123]:
# View the under_30_filter to see the boolean index
under_30_filter

0        True
1        True
2        True
3       False
4       False
        ...  
1333    False
1334     True
1335     True
1336     True
1337    False
Name: age, Length: 1338, dtype: bool

In [125]:
# Get the sum of the true values for the under_30_filter
# This is the number of customers who are under the age of 30
under_30_filter.value_counts()


age
False    921
True     417
Name: count, dtype: int64

In [127]:
# # Use the filter to view only rows with customers who are under 30
df_insurance[under_30_filter]

age     sex     bmi  children  smoker     region  charges
0      19  female  27.900         0       1  southwest  16885.0
1      18    male  33.770         1       0  southeast   1726.0
2      28    male  33.000         3       0  southeast   4449.0
10     25    male  26.220         0       0  northeast   2721.0
12     23    male  34.400         0       0  southwest   1827.0
...   ...     ...     ...       ...     ...        ...      ...
1328   23  female  24.225         2       0  northeast  22396.0
1331   23  female  33.400         0       0  southwest  10796.0
1334   18  female  31.920         0       0  northeast   2206.0
1335   18  female  36.850         0       0  southeast   1630.0
1336   21  female  25.800         0       0  southwest   2008.0

[417 rows x 7 columns]

2) Create a filter for males

In [139]:
# Create a filter for males called male_filter
male_filter=df_insurance['sex']=='male'
male_filter

0       False
1        True
2        True
3        True
4        True
        ...  
1333     True
1334    False
1335    False
1336    False
1337    False
Name: sex, Length: 1338, dtype: bool

In [141]:
# View the filter to see the boolean index
male_filter

0       False
1        True
2        True
3        True
4        True
        ...  
1333     True
1334    False
1335    False
1336    False
1337    False
Name: sex, Length: 1338, dtype: bool

In [143]:
# Obtain the sum of True values in the male_filter
# This is the number of customers who are male
male_filter.value_counts()

sex
True     676
False    662
Name: count, dtype: int64

In [145]:
# Use the filter to view only rows with customers who are male
df_insurance[male_filter]

age   sex     bmi  children  smoker     region  charges
1      18  male  33.770         1       0  southeast   1726.0
2      28  male  33.000         3       0  southeast   4449.0
3      33  male  22.705         0       0  northwest  21984.0
4      32  male  28.880         0       0  northwest   3867.0
8      37  male  29.830         2       0  northeast   6406.0
...   ...   ...     ...       ...     ...        ...      ...
1324   31  male  25.935         1       0  northwest   4240.0
1325   61  male  33.535         0       0  northeast  13143.0
1327   51  male  30.030         1       0  southeast   9378.0
1329   52  male  38.600         2       0  southwest  10325.0
1333   50  male  30.970         3       0  northwest  10601.0

[676 rows x 7 columns]

3) Combine the under_30_filter and the male_filter to create a single filter called under_30_male_filter

In [151]:
# Combine the under_30_filter and the male_filter to create a single filter called under_30_male_filter
under_30_male_filter=under_30_filter & male_filter


In [153]:
# Display the boolean index of the under_30_male_filter
under_30_male_filter

0       False
1        True
2        True
3       False
4       False
        ...  
1333    False
1334    False
1335    False
1336    False
1337    False
Length: 1338, dtype: bool

In [157]:
# Get the sum of the under_30_male_filter True values
# This is the number of customers who are both under 30 and male
under_30_male_filter.sum()

216

In [163]:
# Use the filter to view only rows with customers who are both under 30 and male
df_insurance[under_30_male_filter]


age   sex     bmi  children  smoker     region  charges
1      18  male  33.770         1       0  southeast   1726.0
2      28  male  33.000         3       0  southeast   4449.0
10     25  male  26.220         0       0  northeast   2721.0
12     23  male  34.400         0       0  southwest   1827.0
14     27  male  42.130         0       1  southeast  39612.0
...   ...   ...     ...       ...     ...        ...      ...
1292   21  male  23.210         0       0  southeast   1515.0
1295   20  male  22.000         1       0  southwest   1965.0
1296   18  male  26.125         0       0  northeast   1709.0
1315   18  male  28.310         1       0  northeast  11272.0
1317   18  male  53.130         0       0  southeast   1163.0

[216 rows x 7 columns]

4) Create a filter for regions that contain south using .str.contains



In [168]:
# Create a filter for regions that contain south using .str.contains
south_region=df_insurance['region'].str.contains('south')
south_region

0        True
1        True
2        True
3       False
4       False
        ...  
1333    False
1334    False
1335     True
1336     True
1337    False
Name: region, Length: 1338, dtype: bool

In [170]:
# Use the filter to view only rows with regions containing "south"
df_insurance[south_region]

age     sex    bmi  children  smoker     region  charges
0      19  female  27.90         0       1  southwest  16885.0
1      18    male  33.77         1       0  southeast   1726.0
2      28    male  33.00         3       0  southeast   4449.0
5      31  female  25.74         0       0  southeast   3757.0
6      46  female  33.44         1       0  southeast   8241.0
...   ...     ...    ...       ...     ...        ...      ...
1330   57  female  25.74         2       0  southeast  12629.0
1331   23  female  33.40         0       0  southwest  10796.0
1332   52  female  44.70         3       0  southwest  11412.0
1335   18  female  36.85         0       0  southeast   1630.0
1336   21  female  25.80         0       0  southwest   2008.0

[689 rows x 7 columns]